<launch>

    <node pkg="auv_vision" name="gate_locator" type="gate_locator" output="screen">
        <param name="detectorMode" value="vertical" />
        <param name="horizontalToVerticalRelation" value="1.5" />
        <param name="debugVision" value="false" />
    </node>

    <node pkg="auv_vision" name="drums_locator" type="drums_locator" output="screen">
        <param name="debugVision" value="false" />
    </node>

    <node pkg="auv_pilot" name="pilot" type="pilot" output="screen">
        <param name="simulation" value="false" />
        <param name="velocityLevel1" value="3000.0" />
        <param name="velocityLevel2" value="5000.0" />
        <param name="velocityLevel3" value="7000.0" />
        <param name="velocityLevel4" value="15000.0" />
        <param name="diveTime" value="5.0" />
    </node>

    <node pkg="auv_connection" name="uart_bridge" type="uart_bridge" output="screen" respawn="true" >
        <param name="device" value="/dev/ttyS0" />
        <param name="baudrate" value="57600" />
        <param name="dataBytes" value="8" />
        <param name="parity" value="none" />
        <param name="stopBits" value="1" />
    </node>
    <node pkg="auv_connection" name="hardware_bridge" type="hardware_bridge" output="screen" respawn="true" />

    <!-- First front camera -->
    <node name="cam_front_1" pkg="usb_cam" type="usb_cam_node" output="screen" respawn="true" >
        <param name="video_device" value="/dev/v4l/by-id/usb-3.0_USB_Camera_3.0_USB_Camera-video-index0" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="framerate" value="30" />
        <param name="camera_frame_id" value="cam_front_1" />
        <param name="camera_name" value="cam_front_1" />
        <param name="io_method" value="mmap"/>
    </node>

    <!-- Bottom camera -->
    <node name="cam_bottom" pkg="usb_cam" type="usb_cam_node" output="screen" respawn="true" >
        <param name="video_device" value="/dev/v4l/by-id/usb-Sonix_Technology_Co.__Ltd._H264_USB_Camera_SN0001-video-index0" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="framerate" value="30" />
        <param name="camera_frame_id" value="cam_bottom" />
        <param name="camera_name" value="cam_bottom" />
        <param name="io_method" value="mmap"/>
    </node>

    <node pkg="auv_pilot" name="top_level_fsm" type="top_level_fsm.py" output="screen">
        <param name="startCondition" value="trigger" />
        <param name="mode" value="none" />
        <param name="diveDelay" value="1" />
        <param name="initialDepth" value="50" />
        <param name="imuReset" value="false" />
        <param name="qualificationDuration" value="25" /> <!-- March move duration in seconds -->
    </node>

    <node pkg="zbar_ros" type="barcode_reader_node" name="barcode_reader" output="screen">
        <remap from="image" to="/cam_front_1/image_raw" />
    </node>

    <node pkg="auv_startup" type="qr_trigger.py" name="qr_trigger" />

</launch>